rundeckapp/src/test/groovy/org/rundeck/app/components/JobYAMLFormatSpec.groovy

```diff
@@ -264,6 +264,166 @@ class JobYAMLFormatSpec extends Specification {
         result[0].notification['onsuccess'].findAll{ it['urls'] != null }.size() == 2
     }
 
+    @Unroll
+    def "should return a notification list with one webhook notification"() {
+        given:
+        def input = "" +
+                "- defaultTab: nodes\n" +
+                "  description: ''\n" +
+                "  executionEnabled: true\n" +
+                "  loglevel: INFO\n" +
+                "  name: a\n" +
+                "  nodeFilterEditable: false\n" +
+                "  notification:\n" +
+                "    onsuccess:\n" +
+                "      format: json\n" +
+                "      httpMethod: post\n" +
+                "      urls: http://localhost:4440/project\n" +
+                "  notifyAvgDurationThreshold: null\n" +
+                "  plugins:\n" +
+                "    ExecutionLifecycle: null\n" +
+                "  scheduleEnabled: true\n" +
+                "  schedules: []\n" +
+                "  sequence:\n" +
+                "    commands:\n" +
+                "    - exec: asd\n" +
+                "    keepgoing: false\n" +
+                "    strategy: node-first"
+        def sut = new JobYAMLFormat()
+        when:
+        def result = sut.decode(new StringReader(input))
+        then:
+        result[0].notification['onsuccess'].size() == 1
+        result[0].notification['onsuccess'].findAll{ it['urls'] != null }.size() == 1
+    }
+
+    @Unroll
+    def "should return a notification list of map and single map with email and webhook notifs"() {
```

Comment:
This name sounds confusing.

This test seems to be trying to check that either a list of notifications or a notification map will result on a list of notifications for each trigger always. So I think a more suitable name (thinking on the testcase and not on the exact code) for this test would be something like:
`should return list of notifications for each trigger when some triggers have notifs in yaml list and some not in a list`